x/lockup/keeper/lock.go

```diff
@@ -48,18 +48,21 @@ func (k Keeper) BeginUnlockAllNotUnlockings(ctx sdk.Context, account sdk.AccAddr
 }
 
 // AddToExistingLock adds the given coin to the existing lock with the same owner and duration.
-// Returns an empty array of period lock when a lock with the given condition does not exist.
-func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) ([]types.PeriodLock, error) {
+// Returns an empty struct of period lock when a lock with the given condition does not exist.
+func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) (types.PeriodLock, error) {
 	locks := k.GetAccountLockedDurationNotUnlockingOnly(ctx, owner, coin.Denom, duration)
 	// if existing lock with same duration and denom exists, just add there
 	if len(locks) > 0 {
+		// there should only be a sinlge lock with the same duration + token, thus we take the first lock
 		lock := locks[0]
 		_, err := k.AddTokensToLockByID(ctx, lock.ID, owner, coin)
 		if err != nil {
-			return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+			return types.PeriodLock{}, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
 		}
+		return locks[0], nil
```

Comment:
I have a concern - so we add tokens to lock by id but then return an old lock. Is that expected? 

The lock must be updated with tokens, maybe we need to return a new lock returned by ` AddTokensToLockByID(...)`

Reply:
What if you try logging `updatedLock`? That's what I suspect is going to be outdated.

It seems to me that we update the db correctly so a future
```
lock1, err := suite.App.LockupKeeper.GetLockByID(suite.Ctx, updatedLock.ID)
```

should work as expected. However, the return from `AddToExistingLock` itself might be outdated

Reply:
Here's the output, interesting thing it is logging the same thing when i try it with return value you suggested. 
```
Matt 
Lock: {1 osmo1v9jxgu3395kj6tfd95kj6tfd95kj6tfd0hptum 1s 0001-01-01 00:00:00 +0000 UTC 10stake}
UpdatedLock:  {1 osmo1v9jxgu3395kj6tfd95kj6tfd95kj6tfd0hptum 1s 0001-01-01 00:00:00 +0000 UTC 10stake}

Roman
Lock: {1 osmo1v9jxgu3395kj6tfd95kj6tfd95kj6tfd0hptum 1s 0001-01-01 00:00:00 +0000 UTC 10stake}
UpdatedLock: {1 osmo1v9jxgu3395kj6tfd95kj6tfd95kj6tfd0hptum 1s 0001-01-01 00:00:00 +0000 UTC 10stake}
```

I think a solution might be to return the lock from `TestLog, err := k.AddTokensToLockByID(ctx, lock.ID, owner, coin)` which actually outputs, the correct result: `TestLog: ID:1 owner:"osmo1v9jxgu3395kj6tfd95kj6tfd95kj6tfd0hptum" duration:<seconds:1 > end_time:<seconds:-62135596800 > coins:<denom:"stake" amount:"20" > `

Reply:
Yeah we need to be returning from `AddTokensToLockByID`

Reply:
good catch!

I took a more detailed look, currently changing this to simply return the lock ID, as that's whats only being used when this method is getting called. No reason to pass locks around!